doc/source/how_to/execution/creating_custom_executors.ipynb

```diff
@@ -0,0 +1,481 @@
+    "                calling this function.\n",
+    "            node_id: The node ID of this task in the workflow graph.\n",
+    "\n",
+    "        Returns:\n",
+    "            output: The result of the executed function.\n",
+    "        \"\"\"\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<font size=\"3\">\n",
+    "Here, we convert the function from a Covalent TransportableObject to a \"standard\" Python function. From here, we are free to create whatever operations make this custom executor unique.\n",
+    "</font>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```python\n",
+    "        # Convert the function to be executed, which is in the form of\n",
+    "        # a Covalent TransportableObject, to a 'standard' Python function:\n",
+    "        fn = function.get_deserialized()\n",
+    "        app_log.debug(type(fn))\n",
+    "\n",
+    "        # In this block is where operations specific to your custom executor\n",
+    "        # can be defined. These operations could manipulate the function, the\n",
+    "        # inputs/outputs, or anything that Python allows.\n",
+    "        external_object = ExternalClass(3)\n",
+    "        app_log.debug(external_object.multiplier)\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<font size=\"3\">\n",
+    "The execution of the input function must be done within the following context manager:\n",
+    "</font>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```python\n",
+    "        with self.get_dispatch_context(DispatchInfo(dispatch_id)), redirect_stdout(\n",
+    "            io.StringIO()\n",
+    "        ) as stdout, redirect_stderr(io.StringIO()) as stderr:\n",
+    "            # Here we simply execute the function on the local machine.\n",
+    "            # But this could be sent to a more capable machine for the operation,\n",
+    "            # or a different Python virtual environment, or more.\n",
+    "            result = fn(**kwargs)\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<font size=\"3\">\n",
+    "Other unique custom operations or informational logging can be done here as well.\n",
+    "</font>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```python\n",
+    "        # Other custom operations can be applied here.\n",
+    "        result = self.helper_function(result)\n",
+    "\n",
+    "        debug_message = f\"Function '{fn.__name__}' was executed on node {node_id} with execution arguments {execution_args}\"\n",
+    "        app_log.debug(debug_message)\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<font size=\"3\">\n",
+    "Finally, return the result and any log/error messages, if any:\n",
```

Comment:
As mentioned above, the `stdout` variable will contain the standard out, i.e any print statements done in execution. `stderr` will contain the standard error, so if anything failed on an OS level then it'll be returned, people can also pipe print statement's output to stderr, so we had to take that into consideration as well.

Reply:
Also re-worded, to make clear what the log/error messages contain.